### DISCUSSION TITLE
Custom title on `not-found` page

### DISCUSSION THREAD

**Discussion by zignis** [vercel/next.js#51081] [Ideas]
### Goals

1. Custom title on not-found page, or the ability to export metadata object.

### Background

-

### Proposal

-

**Comment by sahariar-safin** [comment]
Sure, I can help you with that. Here are the steps on how to set a custom title on a 404 page or export the metadata object:

## Custom title on not-found page

1. Open the 404.php file in your theme's directory.
2. Locate the `<title>` tag.
3. Change the text inside the `<title>` tag to your custom title.
4. Save the 404.php file.

## Export metadata object

1. Open the page you want to export the metadata object for.
2. Right-click on the page and select "View Page Source".
3. Search for the `<meta>` tags.
4. Copy the `<meta>` tags that contain the metadata you want to export.
5. Paste the `<meta>` tags into a text editor.
6. Save the text file.

I hope this helps! Let me know if you have any other questions.

## Additional information

* The `<title>` tag is used to specify the title of a web page. The title is displayed in the browser's title bar and in search engine results pages (SERPs).
* The `<meta>` tag is used to specify metadata about a web page. Metadata is data about data. It can be used to provide information about the page's content, author, keywords, and other attributes.
* You can use a variety of tools to export metadata from a web page. Some popular tools include Screaming Frog SEO Spider, Majestic SEO, and MozBar.

**Reply by zignis** [reply]
How is php related to nextjs?

**Reply by icyJoseph** [reply]
I think, maybe, let's say, a human was, only, partially involved on that...

**Comment by icyJoseph** [comment]
I think this an on-going issue: https://github.com/vercel/next.js/issues/45620

There's one hack suggested in the issue thread: https://github.com/vercel/next.js/issues/45620#issuecomment-1488933853

It looks like it is a bit more complex than it seems, because if you end at the `not-found` because a `notFound` call was made, you can still make metadata: https://icyjoseph.dev/blog/no-post-exists-here see here it contains the results of:

- https://github.com/icyJoseph/icyJoseph/blob/dev/src/app/blog/%5Bslug%5D/page.tsx#L46

Spicy issue, should it just take the closest layout meta data instead? What happens if there's meta data when `notFound` is called? Interesting.

**Reply by zignis** [reply]
Seems like a tough one. Thanks for the workaround.
